lib/benchee/system.ex

```diff
@@ -61,19 +61,38 @@ defmodule Benchee.System do
   """
   def cpu_speed, do: cpu_speed(os())
 
-  defp cpu_speed(:macOS), do: system_cmd("sysctl", ["-n", "machdep.cpu.brand_string"])
   defp cpu_speed(:Windows), do: "N/A"
-  defp cpu_speed(:Linux), do: "N/A"
+  defp cpu_speed(:macOS), do: system_cmd("sysctl", ["-n", "machdep.cpu.brand_string"])
+  defp cpu_speed(:Linux) do
+    ["model name\t:" <> cpu_info] = linux_cmd("/proc/cpuinfo", ~r/model name.*:[\w \(\)\-\@\.]*ghz/i)
+    String.trim(cpu_info)
+  end
 
   @doc """
   Returns an integer with the total number of available memory on the machine
   running the benchmarks.
   """
   def available_memory, do: available_memory(os())
 
-  defp available_memory(:macOS), do: system_cmd("sysctl", ["-n", "hw.memsize"])
   defp available_memory(:Windows), do: "N/A"
-  defp available_memory(:Linux), do: "N/A"
+  defp available_memory(:macOS) do
+    {memory, _} = Integer.parse(system_cmd("sysctl", ["-n", "hw.memsize"]))
+    format_memory(memory, 1_000_000_000)
+  end
+  defp available_memory(:Linux) do
+    ["MemTotal:" <> memory] = linux_cmd("/proc/meminfo", ~r/MemTotal.*kB/)
+    {memory, _} = memory
+                  |> String.trim()
+                  |> String.trim_trailing(" kB")
+                  |> Integer.parse
+    format_memory(memory, 1_000_000)
+  end
+
+  defp format_memory(memory, coefficient), do: "#{memory / coefficient} GB"
+
+  defp linux_cmd(file, regex) do
+    Regex.run(regex, system_cmd("cat", [file]))
+  end
```

Comment:
Yeah, I'll sleep on it tonight and tomorrow morning hopefully something great will come to me. I'm thinking maybe `parse_cmd_result` or something like that? Frankly, I'm not even 100% sure that the function justifies its existence since it's so small, but we'll see how I feel about it tomorrow when I look at it with fresh eyes.